app/actions/views/channel.js

```diff
@@ -673,6 +675,8 @@ export function loadChannelsForTeam(teamId, skipDispatch = false) {
 
                 // Fetch needed profiles from channel creators and direct channels
                 dispatch(loadSidebar(data));
+
+                dispatch(loadUnreadChannelPosts());
```

Comment:
you should pass `data` here and rely on the fetched results instead of what is in the store as we do for the sidebar